dex/src/state.rs

```diff
@@ -3136,9 +3251,15 @@ impl State {
         if eq_buf.len() < 128 {
             Err(DexErrorCode::EventQueueTooSmall)?
         }
+        let account_flags = if consume_events_authority.is_some() {
+            (AccountFlag::Initialized | AccountFlag::EventQueue | AccountFlag::IncludeTakerMetadata)
```

Comment:
Given that `IncludeTakerMetadata` can only be used with a consume events atuhority, we can remove `IncludeTakerMeatadata` and just re-use the `CrankAuthorityRequired` flag.